src/utils/permissions.ts

```diff
@@ -191,8 +112,134 @@ export async function hasChannelPermissions(
   if (permissions.every((perm) => allowedPermissions.has(perm))) return true;
 
   // Some permission was not explicitly allowed so we default to checking role perms directly
-  const hasPerms = await botHasPermission(guild.id, permissions);
-  return hasPerms;
+  return await calculateServerPermissions(memberID, guild.id, permissions);
+}
+
+export async function calculateServerPermissions(
+  memberID: string,
+  guildID: string,
+  permissions: Permission[],
+): Promise<true | "GUILD_NOT_FOUND" | "MEMBER_NOT_FOUND" | Permission> {
```

Comment:
Please remove the return type, they are extremely problematic and can cause impossible to solve bugs